src/factories/nodes.ts

```diff
@@ -64,18 +65,20 @@ export async function nodesContainerFactory() {
     highlightSelectedNode()
   })
 
-  async function render(data: RunGraphData): Promise<void> {
+  async function render(data: DataBundle): Promise<void> {
     startWorker()
 
-    runData = data
+    dataBundle = data
     nodesLayout = null
 
     await Promise.all([
-      createNodes(data),
-      createEdges(data),
+      createNodes(),
+      createEdges(),
```

Comment:
By not passing data into these methods now the methods have to check if data is defined. So I'd switch that back (because we know the data exists here where they are called) and just pass in the specific piece of data each method needs.